python/packages/ai/teams/ai/models/openai_model.py

```diff
@@ -151,6 +188,14 @@ async def complete_prompt(
 
         messages: List[chat.ChatCompletionMessageParam] = []
 
+        # Submit tool outputs, if previously invoked
+        if memory.has(ACTIONS_HISTORY):
+            tools_messages = cast(
+                List[chat.ChatCompletionMessageParam], memory.get(ACTIONS_HISTORY)
+            )
+            messages.extend(tools_messages)
+            memory.set(ACTIONS_HISTORY, [])
+
         for msg in res.output:
```

Comment:
doesn't this for loop need to be updated to support tool messages?

Reply:
nope, we're propagating it over in `L192-197` via the changes in action_planner